### ISSUE TITLE
combine harvester - cutter uses too much power

### ISSUE THREAD

**Issue by mieperDE** [Mogli12/GearboxAddon#114]
The combine harvesters are very weak on the mountain. Is this really so extreme in reality?
Some harvesters are only by the cutting mechanism (without driving) already on 100% load.
600PS can't get faster than 5 km/h on a small slope. The average speed is ~3km/h.

There is no difference between a 200ps and a 600ps combine harvester.

**Comment by Mogli12** [comment]
I cannot reproduce this without an example. And which version of the gearbox addon do you use? But the real problem is that a combine harvester does not consume any power. It does not even have the PowerConsumer spec.

**Comment by mieperDE** [comment]
Used version: 2.0.4.6

https://youtu.be/GJnVPy01ejE
When I reached the 7 km/h limit, I've changed the gear to 2. But then the machine almost stop.

**Comment by Mogli12** [comment]
How big is this cutter? The power consumption is calculated from the size of the area the cutter worked on. The bigger the cutter the higher the power consumption. I do not think that the 9230 can handle more then 45ft. It only has 571 HP.

**Comment by mieperDE** [comment]
The original Giants CaseIH 3162 Draper 45ft

**Comment by Mogli12** [comment]
To me this looks quite steep. I think the speed is ok. Why don't you disable the gearbox? You can enable and disable this for every vehicle with KEY_rshift KEY_semicolon.

**Comment by mieperDE** [comment]
Good point.
I don't know how a Combine Harvester drive in reality. Do you know it?

**Comment by Mogli12** [comment]
No, I do not know it. And I am not even sure if power is really limiting the speed of the harvester. Or does missing power lead to loss and the drive will slow down to avoid loss.

The standard combine with standard transmission does not even consume any power. 

So it is not that I am not willing to fix this. I am just not sure what the "correct" power consumption could be.

**Comment by Mogli12** [comment]
Hallo,

ja, ich spreche eindeutig besser Deutsch als Englisch. Ich habe den Artikel mal überflogen, bin aber nicht wirklich schlau daraus geworden.

Viele Grüße, Stefan

**Comment by mieperDE** [comment]
In einige Foren habe ich von einem Verhältnis von 2/3 gelesen.
Also 66% für den Mähdrescher und 33% zum fahren.

**Comment by reinholdmaehler** [comment]
Hallo Stefan,

ich arbeite bei einem Landmaschinenhersteller und war eine Zeit lang Produktmanager für Mähdrescher. Mein Bereich umfasste grob Maschinen ab 300PS.
Es gibt im Wesentlichen 3 Arten von Mähdreschern - Schüttler, Hybridabscheidung und Rotorabscheidung. Ich glaube das ist aber nicht ganz so wichtig.

Grundsätzlich kann man sagen, dass so ein Fahrantrieb hydrostatisch arbeitet, meistens mit fixem Motor und variabler Pumpe. Die großen MD haben oft variable Pumpe und Motor, um z.B. eine Art Overdrive auf der Strasse und höhere Zugkräfte im Acker hinzukriegen. In der Regel wiegt ein MD leer um die 12-18 Tonnen, i.d.R. hat man 6-10 Tonnen Zuladung + Schneidwerk/Vorsatz. Die Maschinen die ich Betreut habe lagen bei ca. 100PS im Fahrantrieb. Es gibt ein "Zugkraft zu Gesamtgewichtverhältnis" von ca. 1,2 bi 1,5.
Die Motoren stehen oft bei Nenndrehzahl, werden so um 300Umin gedrückt bei der Arbeit, was dann 100% Auslastung ist. Langsamer darf es nicht werden, da sonst die Aggregate nicht mehr mithalten können. Der Fahrantrieb hat i.d.R. nur bei starker Bergauffahrt echte Auswirkungen auf die Motorauslastung. Leerlaufdrehzahl ist oft bei 12-1400Umin um Lüfter / Aggregate im Leerlauf für die Warmen Sommertemparaturen entsprechend drehen zu lassen.

Der Strohhäcksler bei einem 12m Schneidwerk im Weizenstroh geht hoch auf 80-100PS, mit Variant um die 20% nach oben und unten - vom Durchsatz abhängig. Ich denke das könnte man an die Liter per qm x Flächenleistung binden.

Im Weizen als "anspruchsvolle" Fruchtart kann ein 600PS MD locker 60-80t die Stunde ernten, da auch viel Stroh durch das System muss. Als Extremfrucht "Mais" mit "fast keinem Stroh" im Gutfluss habe ich bis zu 150t/Stunde bei 12m breite und 600PS gesehen. Aus Spielspasssicht sollte nach Ertragshöhe/Spiellevel evtl. ein Faktor bei den o.g. Tonnen/Stunden überlegt werden. 

Das Abtankrohr braucht pauschal 5-10% Motorleistung wenn im Einsatz. Bei der Fahrt wird z.T. je nach Hersteller durch "Motorboost" oder "langsamer Fahren" dies ausgeglichen. Im Stand nicht.

Ein sauber eingestellter Mähdrescher fährt bei 2% Verlusten und 100% Motorauslastung die o.g. Durchsätze. Wenn man Verluste mit betrachtet, kann es schnell äusserst komplex werden. Ich denke die Nutzer von deinem Addon können da nach eigenem Ermessen die Geschwindigkeit rausnehmen.

Vorsatzgeräte für Getreide liegen immer so zwischen 20-50PS, wobei der Unterschied zwischen "Bandzuführung" und "Schneckenzuführung" sich dahingegen zeigen, dass Bandschneidwerke eher konstante und Schneckenschneidwerke geschwindigkeitsabhängige (proportional) Leistungsaufnahme haben.
Maisschneidwerke sind i.d.R. bei 20PS je Reihe, oft haben wir 75cm Reihenabstand, zumindest ist es eine sehr gängige breite.

Als Mähdrescherspezifische Gearboxparameter zur Einstellung könnte ich mir vorstellen:
- max power for straw chopper (cool wäre, wenn du den "Strohhäckslerstatus" einbeziehst)
- max power for unloading auger
- max power ground drive (irgendwann geht bei z.B. 400bar das Überdruckventil auf)
- min power for System at idle (muss nicht, kann)
- max througput t/hour grain (Durchsatz) at 100% engine load
- max througput t/hour maize (Durchsatz) at 100% engine load
- max power for Header, constantpower=true/false bzgl. Art (keine Ahnung ob es Sinn macht für das Schneidwerk ein eigenes Vehicle anzulegen) 

Das Fahrverhalten ist an sich immer so, das die Geschwindigkeitsänderungen zügig passieren, es schwankt eher die Motordrehzahl durch Lastspitzen im Gutfluss oder "Vollgas den Berg hoch" als die Fahrgeschwindigkeit.

Ich hoffe das könnte dir eine erste Idee geben.
Gruß
Reinhold

**Comment by yumi-modding** [comment]
Hello Reinhold,
Did you have a look at moreRealistic mod ? I personally think it's a lot better than genuine giants engine (mgmt of different power consumed by the combine+header) based on input material. I don't really think it's useful to have a 3 or 4 gears gearbox on a combine in game and find continuous transmission OK instead.

**Comment by Mogli12** [comment]
Hallo Reinhold,

vielen Dank für diese super Informationen. Ich werde mal versuchen, es umzusetzen.

@yumi-modding: Yes, I know moreRealistic and if moreRealistic is present I turn off the combine power calculation in the gearbox. But since a combine does not consume any power in the vanilla game it is still necessary to think about it. 

Viele Grüße, Stefan

**Comment by reinholdmaehler** [comment]
@yumi-modding:
yes I know about more realistic and it safed FS13 for me. The gearbox Addon moved me to FS15 and then to FS17. I want to switch to more realistic, but it is a considerable amount of work to adjust the mods I use to MR. However, I create my own gearbox profiles for all mods I use (and am fairly fast by now). I dont know if MR allows for gears or just CVT - if it is all cvt I am a lot less interested.

**Comment by reinholdmaehler** [comment]
Hallo Stefan,

das freut mich. Was mir noch eingefallen ist:
John Deere bietet für Erntemaschinen ein 2 Gang Getriebe an, das mit einem Hydrostaten kombiniert ist. Die zwei Gänge werden "unter Last" geschaltet, um so etwas ähnliches wie CVT zu haben. CLAAS, New Holland, Case IH und Rostelmasch können "nur" im Stillstand unsynchronisiert geschaltet werden. Bei z.B. NH/Case wird zum Schalten die Parkbremse aktiviert, bei CLAAS muss die Fussbremse betätigt sein. Die Gänge sind i.d.R. unsynchronisiert, das Schalten wird durch "Ruckeln" des Hydromotors unterstützt.

Wenn du Interesse an einem Profil für Feldhäcksler hast, kann ich ggf. auch helfen.
Gruß
Reinhold

**Comment by Mogli12** [comment]
Hallo Reinhold,

dass man Gänge nur im Stand schalten kann, das ist ohnehin nützlich. Ich probiere mal was.

Viele Grüße, Stefan

**Comment by Mogli12** [comment]
Hallo Reinhold,

derzeit berechnet das GB AddOn die Leistungsaufnahme ohne moreRealsitic folgendermaßen:

Es gibt 5 Parameter 
- unloadingPowerConsumption in kN; zus. Kraft fürs Abbunkern
- threshingPowerConsumption in kN; Kraft im Leerlauf
- threshingPowerConsumptionInc in kN * s / l, also Kraft pro l/s Durchsatz
- chopperPowerConsumption in kN; nur für den Strohhäcksler
- chopperPowerConsumptionInc in kN * s / l, also Kraft pro l/s Durchsatz

Aber die Werte setzt eigentlich niemand explizit. Jetzt gibt es 2 Möglichkeiten für die Defaulwerte:
- Maishäcksler ohne Angabe der Breite
  -  unloadingPowerConsumption = 0
  - threshingPowerConsumption = 0.3 * maxTorque
  - threshingPowerConsumptionInc = 9.2
  - chopperPowerConsumption = 0
  - chopperPowerConsumptionInc = 0
- Mähdrescher ohne Angabe der Breite
  -  unloadingPowerConsumption = 0
  - threshingPowerConsumption = 0.5 * maxTorque
  - threshingPowerConsumptionInc = 6
  - chopperPowerConsumption = 3.2 * maxTorque
  - chopperPowerConsumptionInc = 0.383
- Maishäcksler mit Angabe der Breite in Metern
  -  unloadingPowerConsumption = 0
  - threshingPowerConsumption = Breite * 12
  - threshingPowerConsumptionInc = 9.2
  - chopperPowerConsumption = 0
  - chopperPowerConsumptionInc = 0
- Mähdrescher mit Angabe der Breite in Metern
  -  unloadingPowerConsumption = 6 + Breite
  - threshingPowerConsumption = Breite * 10
  - threshingPowerConsumptionInc = 6
  - chopperPowerConsumption = Breite * 0.64
  - chopperPowerConsumptionInc = 0.383

Diese Werte gelten zunächst mal für 7 km/h. Nun wird das ganze noch auf die Wunschgeschwindigkeit (Default derzeit 10 km/h) skaliert, also mit 0.7 multipliziert.

Jetzt wäre es mal interessant, wenn man meine Defaults mal mit echten Werten vergleichen könnte.

Das Schalten der Gänge nur im Stand ist übrigens in der aktuellen Version drin. Probier einfach mal den Case IH 1660. Beim dem Drescher habe ich übrigend die Breite auf 4,3 Meter eingestellt.

Viele Grüße, Stefan

**Comment by reinholdmaehler** [comment]
Hallo Stefan,

ok, cool. Habe diese "default" Werte bzw. die Kategorien in deiner lua / xml geshen - jetzt weiß ich auch warum ich mit den Tier4 LEXIONs immer unzufrieden war: threshingPowerConsumption = Breite * 10. Das hatte ich so nicht erwartet....
Ich guck mal ob ich in unserer Firma an solche Leistungsmessdaten dran komme, vermute mal nicht. 
Was ich machen kann ist ein paar Messschriebe von einem Mähdrescher / Feldhäcksler aus unserem Telemetriesystem zu ziehen. Man kann zwar nicht die Einzelaggregate rauslesen, jedoch z.B. den Momentankraftstoffverbrach bzw Motorauslastung und Drehzahl.

Ich probier das mal mit dem alten Axialflowl....

Gruß
Reinhold

**Comment by messi10no1** [comment]
Hey! I have same problem. My harvester can't move normal speed, just 4-5km/h maybe 6. Doesnt matter 200 or 500hp. Always flashing the high PTO torque warning, and after that combine stopping. I feel the header eat too much torque than normal. I want to switch off gearbox ONLY for combines, but doens't work at all. I have 2 options, switch off ALL, and everytime i want to switching to tractors I must to turn ON and OFF and ON and OFF. I can't do this all time.. This is boring. But if I could turn off separately, this will be the second best solution.

**Comment by Mogli12** [comment]
Hello Tom!
You can turn off the gearbox in the configuration menu of your combine. The entry at the bottom is for the gearbox. Please select "off" and everything should be fine.
Regards, Stefan

**Comment by messi10no1** [comment]
Hello Stefan!
Thank you for the asnwer! I've already tried this. But unfortunately, if I turn it off, it will freeze everything. I can't move, the time is moving, but nothing happen. Lost keyboard and everything. Once the harvester started jumping and flying and fly to my workshop's top. :D That was funny and sad same time..
I tried this on mod harvester and factory harvester also. I don't understand.. 
Tom

**Comment by Mogli12** [comment]
Hello Tom!
Can you post your log.txt in [gist](https://gist.github.com/)? I heard about it but was never able to reproduce it.
Regards, Stefan

**Comment by messi10no1** [comment]
Hello Stefan!
I try copy the log soon. Meanwhile I tried i'm trying to narrow down the possible problem mods. (sorry for my english) Almost everytime the harvester duplicated! Wierd.

**Comment by Mogli12** [comment]
Hello Tom!
I do not use better fuel usage but more realistic. That explains why I was not able to reproduce it.
Thanks, Stefan

**Comment by messi10no1** [comment]
Hello Stefan!
I using this mod since FS15. But I never had a problem cause this mod.. At least I did not know about it. :) I had a similar problem due to the ground modifier roller. Freeze the game. But that mod is now perfect. By the way I thank you too! :) 
Best Regards, Tom

**Comment by Mogli12** [comment]
Hello Tom!
Version 3.1 build 38 should avoid the log errors in betterFuelUsage.lua.
Regards, Stefan

**Comment by messi10no1** [comment]
Hello Stefan!
Thank you! This version is working fine!

Regards, Tom

**Comment by Helgomant** [comment]
Hallo Stefan,

ich möchte das Thema Hydrostat in den Erntemaschinen noch mal aufgreifen.
Ich nutze das Gb Addon in der Version 3.1 und habe änhnliche Erfahrungen gemacht, wie hier bereits beschrieben. Ich spiele im Moment mit dem Moddrescher CLAAS Dominator 88S, habe per Mod realistische Ernteerträge und Gewichte der Fruchtsorten hinzugefügt. Mit einem normalen Schneidwerk für Getreide erreiche ich an Steigungen teilweise keine 6km/h (schneller mähe ich nicht) mehr. Selbst wenn ich nur mit eingeschaltetem Dreschwerk die Steigung hochfahre (ohne zu mähen) braucht der Drescher ewig um auf die Geschwindigkeit zu kommen. 

In Missionen (blödes Zeitlimit) fahre ich schneller. Auch hier habe ich die Erfahrung gemacht, das an Steigungen bei ca. 6 bis 7 km/h Schluss ist. Auch mit den großen Dreschern. 

Mit Maisgebiss am Dominator 88S wird das ganze noch schlimmer. Es sieht so aus, als ob das Gebiss minimalen Bodenkontakt hat, sodass in der Ebene bei 4 km/h Ende ist. Zudem bekomme ich im Mais auch oftmals die Drehmoment Fehlermeldung. Das selbe auch bei Missionen mit den großen Dreschern bergab, da dort die Geschwindigkeit größer ist und somit der Durchsatz steigt. 

Der Grimme Tectron ist mit den default Einstellungen im Gb Addon kaum fahrbar. In der Ebene ist bei ca. 3 bis 4 km/h Schluss (so sind keine Missionen zu schaffen). Auch hier kommt oftmals der Drehmoment Fehler, obwohl ich schon mit reduzierten Erntemengen bei Kartoffeln spiele.

Zudem ist mir aufgefallen, das die Motorlast beim Ernten (ca. 200min-1 unter der max. Drehzahl) nie auf 100% kommt!? Woran liegt das?

Mir kommt es so vor, als ob der Hydrostat Antrieb in den Erntemaschinen nur für die Bewegung deutlich zu viel Energie verbraucht, sodass am Ende nicht mehr genug Antriebsleistung für die Dresch- und Rodeaggregate übrig ist bzw. der Wirkungsgrad Antrieb zu Geschwindigkeit zu schlecht ist. Ich bin selbst jahrelang Mähdrescher gefahren (auch mit Hydrostat) und kann sagen, das die Maschinen äusserst spontan auf Geschindigkeitsänderungen mit dem Fahrhebel reagieren. Eine mit einem Hydrostaten angetriebene Erntemaschine würde an Steigungen nahezu keine Geschwindigkeit aufgrund des Antriebes verlieren. Entweder würden die Räder durchdrehen, oder die Maschine fährt mit unverminderter Geschwindigkeit weiter. Selbst im großen Gang ist es kein Problem mit einer 10t Maschine plus vollem Korntank auf dem Feld anzufahren und auch ordentlich zu beschleunigen.

Vielleicht kannst du dir das ja mal anschauen und eventuell gibt es eine Lösung für das Problem.

Mir ist noch ein Sache aufgefallen. Oben hast du beschrieben, wie die Leistung bei Erntemaschinen aufgeteilt wird. Dort hast du die Einheit KN verwendet. Auch im Leerlauf. Hast du die Einheiten für den Leerlauf Leistungsverbrauch nochmal geändert? Denn im Code steht der von dir angegebene Wert x 100!? Wenn das noch KN sind, ist das doch ein bisschen sehr viel, oder?

Viele Grüße und weiter so.
Das ist ganz großen Kino, was du hier machst!

Helge

**Comment by Mogli12** [comment]
Hi,

It seems that in newer version of LS17 the cutter now consumes power. As a result the combine needs twice the power. With the coming fix GearboxAddon will not overwrite the power consumed and only use its own values.

Best regards,
Stefan

Hi,

Es scheint, dass in neueren LS17 Versionen die Schneidwerke Leistung benötigen. Daher benötigt nun der Drescher die doppelte Leistung. Mit der nächsten Korrektur wird GearboxAddon die benötigte Leistung überschreiben und nur den selbst berechneten Wert verwenden.

Viele Grüße,
Stefan

**Comment by RobertLidstroem** [comment]
Super news Stefan! Much appreciated! Thank you!

Do you think you will have the time and willingnes to continue for LS19?
